aptos-move/aptos-vm/src/data_cache.rs

```diff
@@ -278,33 +277,208 @@ impl<S: StateView> AsMoveResolver<S> for S {
-        &self.resource_group_view
+// Manually implement StateStorageView for ExecutorViewProxy - delegate to executor_view_without_group_view
+impl<'a, 'e, E: ExecutorViewWithoutGroupView> StateStorageView for ExecutorViewProxy<'a, 'e, E> {
+    type Key = StateKey;
+
+    fn id(&self) -> StateViewId {
+        self.executor_view_without_group_view.id()
+    }
+
+    fn read_state_value(&self, state_key: &Self::Key) -> Result<(), StateViewError> {
+        self.executor_view_without_group_view.read_state_value(state_key)
+    }
+
+    fn get_usage(&self) -> Result<StateStorageUsage, StateViewError> {
+        self.executor_view_without_group_view.get_usage()
+    }
+}
+
+// Manually implement TResourceGroupView for ExecutorViewProxy - delegate to resource_group_view
+impl<'a, 'e, E: ExecutorViewWithoutGroupView> TResourceGroupView for ExecutorViewProxy<'a, 'e, E> {
+    type GroupKey = StateKey;
+    type ResourceTag = StructTag;
+    type Layout = MoveTypeLayout;
+
+    fn is_resource_groups_split_in_change_set_capable(&self) -> bool {
+        self.resource_group_view.is_resource_groups_split_in_change_set_capable()
+    }
+
+    fn resource_group_size(&self, group_key: &Self::GroupKey) -> PartialVMResult<ResourceGroupSize> {
+        self.resource_group_view.resource_group_size(group_key)
+    }
+
+    fn get_resource_from_group(
+        &self,
+        group_key: &Self::GroupKey,
+        resource_tag: &Self::ResourceTag,
+        maybe_layout: Option<&Self::Layout>,
+    ) -> PartialVMResult<Option<Bytes>> {
+        self.resource_group_view.get_resource_from_group(group_key, resource_tag, maybe_layout)
+    }
+
+    fn resource_size_in_group(
+        &self,
+        group_key: &Self::GroupKey,
+        resource_tag: &Self::ResourceTag,
+    ) -> PartialVMResult<usize> {
+        self.resource_group_view.resource_size_in_group(group_key, resource_tag)
+    }
+
+    fn resource_exists_in_group(
+        &self,
+        group_key: &Self::GroupKey,
+        resource_tag: &Self::ResourceTag,
+    ) -> PartialVMResult<bool> {
+        self.resource_group_view.resource_exists_in_group(group_key, resource_tag)
+    }
+
+    fn release_group_cache(
+        &self,
+    ) -> Option<HashMap<Self::GroupKey, BTreeMap<Self::ResourceTag, Bytes>>> {
+        self.resource_group_view.release_group_cache()
+    }
+}
+
+// Allows to extract the view from `StorageAdapter`. This implementation is critical for the
+// AptosMoveResolver trait, which requires an AsExecutorView implementation.
+//
+// Note that StorageAdapter itself is NOT an ExecutorView - it's a holder that properly dispatches
+// queries. This intentionally forces clients to be explicit about how they're accessing the 
+// underlying views and avoids unintended use.
+impl<'e, E: ExecutorViewWithoutGroupView> AsExecutorView for StorageAdapter<'e, E> {
+    fn as_executor_view(&self) -> &dyn ExecutorView {
+        // TODO: Fix this memory allocation.
```

Comment:
Needs a fix, @georgemitenkov were you suggested there should be a way to pass the storage adapter directly?